docs/usage/CUSTOMER_SYNC.md

```diff
@@ -0,0 +1,201 @@
+````
+    
+##### 2. `warningCallback`
+A callback that is called whenever a warning event occurs during the sync process. Each resource executes its own 
+warning-callback. When sync process of particular resource runs successfully, it is not triggered. It contains the 
+following context about the warning message:
+
+* sync exception
+* customer draft from the source 
+* customer of the target project (only provided if an existing cart discount could be found)
+
+##### Example 
+````java
+ final Logger logger = LoggerFactory.getLogger(CustomerSync.class);
+ final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder
+         .of(sphereClient)
+         .warningCallback((syncException, draft, customer, updateActions) -> 
+            logger.warn(new SyncException("My customized message"), syncException)).build();
+````
+
+##### 3. `beforeUpdateCallback`
+During the sync process if a target customer and a customer draft are matched, this callback can be used to 
+intercept the **_update_** request just before it is sent to commercetools platform. This allows the user to modify 
+update actions array with custom actions or discard unwanted actions. The callback provides the following information :
+ 
+ * customer draft from the source
+ * customer from the target project
+ * update actions that were calculated after comparing both
+
+##### Example
+````java
+final TriFunction<List<UpdateAction<Customer>>, CustomerDraft, Customer,
+                  List<UpdateAction<Customer>>> beforeUpdateCallback, =
+            (updateActions, newCustomerDraft, oldCustomer) ->  updateActions
+                    .stream()
+                    .filter(updateAction -> !(updateAction instanceof SetLastName))
+                    .collect(Collectors.toList());
+                        
+final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder
+                    .of(CTP_CLIENT)
+                    .beforeUpdateCallback(beforeUpdateCallback)
+                    .build();
+````
+
+##### 4. `beforeCreateCallback`
+During the sync process if a cart discount draft should be created, this callback can be used to intercept 
+the **_create_** request just before it is sent to commercetools platform.  It contains following information : 
+
+ * customer draft that should be created
+ ````java
+
+ final Function<CustomerDraft, CustomerDraft> draftFunction =
+             customerDraft -> CustomerDraftBuilder.of(customerDraft)
+                                                  .key(customerDraft.getKey() + "_filteredKey")
+                                                  .build();
+ 
+final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder
+                                                  .of(CTP_CLIENT)
+                                                  .beforeCreateCallback(draftFunction)
+                                                  .build();
+````
```

Comment:
Here we discussed with Roman a while ago. It is not easy to list out a practical example for beforeCreateCallback. We can follow the approach in other resources documents
https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/CATEGORY_SYNC.md#4-beforecreatecallback

Reply:
So I just add "Please refer to the [example in the product sync document](https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/PRODUCT_SYNC.md#example-set-publish-stage-if-category-references-of-given-product-draft-exists)." like in Category sync?

Reply:
yes :)

Reply:
0f595cbd